src/pal/inc/pal.h

```diff
@@ -2589,10 +2592,10 @@ typedef struct _CRITICAL_SECTION {
 
     BOOL bInternal;
     volatile DWORD dwInitState;
-    union CSNativeDataStorage
+    struct CSNativeDataStorage
     {
         BYTE rgNativeDataStorage[PAL_CS_NATIVE_DATA_SIZE]; 
-        VOID * pvAlign; // make sure the storage is machine-pointer-size aligned
+        BYTE rgExtraStorage[CS_NATIVE_DATA_STRUCT_SIZE - PAL_CS_NATIVE_DATA_SIZE]; // make sure the storage is machine-pointer-size aligned
```

Comment:
I've recovered `pal/inc/pal.h` so `CS_NATIVE_DATA_STRUCT_SIZE` is not used any more.